pandas/parser.pyx

```diff
@@ -1151,8 +1181,15 @@ cdef class TextReader:
             # unicode variable width
             return self._string_convert(i, start, end, na_filter,
                                         na_hashset)
-
-
+        # is this comparison good enough?
+        elif dtype == '|O08':
```

Comment:
hmm, don't you have the original (user passed dtype) ? e.g. ``category``? this is very problematic to do this